File: 프로그래머스/lv2/42885. 구명보트/구명보트.java
import java.util.*;
class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        boolean ride = false;
        Arrays.sort(people);
        int i=0, j=people.length-1;
        
        while(i<=j){
            if(people[i]+people[j]<=limit){
                i++;
                j--;
            }
            else{
                j--;
            }
            answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    start,end=0,len(people)-1
    people.sort()
    
    while end>=start:
        if start!=end and people[start]+people[end]<=limit:
            start+=1
        answer+=1
        end-=1
            
    return answer